### PR TITLE
Update references from `master` to `main`

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

- Closes https://github.com/apache/arrow-rs/issues/2173

# Rationale for this change
As @tustvold noticed, there are still some references to the `master` branch which we also need to migrate to `main`

# What changes are included in this PR?

Change all references I could find from `master` to `main`

# Are there any user-facing changes?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,8 +19,6 @@
 
 # Native Rust implementation of Apache Arrow and Apache Parquet
 
-[![Coverage Status](https://codecov.io/gh/apache/arrow-rs/rust/branch/master/graph/badge.svg)](https://codecov.io/gh/apache/arrow-rs?branch=master)
```
Comment (alamb) [inline README.md:22]: We don't run the coverage job anymore, so let's not feature an out of date result prominently:
- https://github.com/apache/arrow-rs/pull/6602

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -138,7 +138,7 @@ cargo test
 cargo test -p arrow
 ```
 
-For some changes, you may want to run additional tests. You can find up-to-date information on the current CI tests in [.github/workflows](https://github.com/apache/arrow-rs/tree/master/.github/workflows). Here are some examples of additional tests you may want to run:
+For some changes, you may want to run additional tests. You can find up-to-date information on the current CI tests in [.github/workflows](https://github.com/apache/arrow-rs/tree/main/.github/workflows). Here are some examples of additional tests you may want to run:
```
Comment (alamb) [inline CONTRIBUTING.md:141]: The old (`master`) urls still work too, but I think it is cleaner to simply update them all

File: parquet/examples/async_read_parquet.rs [hunk 3]
```diff
@@ -45,7 +45,7 @@ async fn main() -> Result<()> {
     builder = builder.with_projection(mask);
 
     // Highlight: set `RowFilter`, it'll push down filter predicates to skip IO and decode.
-    // For more specific usage: please refer to https://github.com/apache/arrow-datafusion/blob/master/datafusion/core/src/physical_plan/file_format/parquet/row_filter.rs.
+    // For more specific usage: please refer to https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/src/physical_plan/file_format/parquet/row_filter.rs.
```
Comment (alamb) [inline parquet/examples/async_read_parquet.rs:48]:
```suggestion
    // For more specific usage: please refer to https://github.com/apache/datafusion/blob/main/datafusion/core/src/datasource/physical_plan/parquet/row_filter.rs.
```

File: parquet/examples/async_read_parquet.rs [hunk 4]
```diff
@@ -45,7 +45,7 @@ async fn main() -> Result<()> {
     builder = builder.with_projection(mask);
 
     // Highlight: set `RowFilter`, it'll push down filter predicates to skip IO and decode.
-    // For more specific usage: please refer to https://github.com/apache/arrow-datafusion/blob/master/datafusion/core/src/physical_plan/file_format/parquet/row_filter.rs.
+    // For more specific usage: please refer to https://github.com/apache/datafusion/blob/main/datafusion/core/src/datasource/physical_plan/parquet/row_filter.rs.
```
Comment (alamb) [inline parquet/examples/async_read_parquet.rs:48]: code moved, so I updated the link

File: object_store/dev/release/README.md [hunk 5]
```diff
@@ -27,11 +27,11 @@ This file documents the release process for the `object_store` crate.
 At the time of writing, we release a new version of `object_store` on demand rather than on a regular schedule.
 
 As we are still in an early phase, we use the 0.x version scheme. If any code has 
-been merged to master that has a breaking API change, as defined in [Rust RFC 1105]
+been merged to main that has a breaking API change, as defined in [Rust RFC 1105]
 the minor version number is incremented changed (e.g. `0.3.0` to `0.4.0`). 
 Otherwise the patch version is incremented (e.g. `0.3.0` to `0.3.1`).
 
-[Rust RFC 1105]: https://github.com/rust-lang/rfcs/blob/master/text/1105-api-evolution.md
+[Rust RFC 1105]: https://github.com/rust-lang/rfcs/blob/main/text/1105-api-evolution.md
```
Comment (mbrobbel) [inline object_store/dev/release/README.md:34]: This is still `master`:
```suggestion
[Rust RFC 1105]: https://github.com/rust-lang/rfcs/blob/master/text/1105-api-evolution.md
```

**Review by mbrobbel — APPROVED** [review]

**Review by tustvold — APPROVED** [review]

**Comment by alamb** [comment]
Thank you @tustvold  and @mbrobbel

**Pull request merged**